# Provide contentLength for download clients.

# NOTE: It might fail sometimes.
--- mozilla/uriloader/exthandler/nsIExternalHelperAppService.idl.orig	2007-03-07 16:22:50.000000000 -0400
+++ mozilla/uriloader/exthandler/nsIExternalHelperAppService.idl	2007-03-07 16:23:20.000000000 -0400
@@ -116,16 +116,21 @@ interface nsIHelperAppLauncher : nsICanc
   readonly attribute nsIURI source;
 
   /**
    * The suggested name for this file
    */
   readonly attribute AString suggestedFileName;
 
   /**
+   * The content length for this file. It might not be available it some cases.
+   */
+  readonly attribute PRInt64 contentLength; 
+
+  /**
    * Called when we want to just save the content to a particular file.
    * NOTE: This will release the reference to the nsIHelperAppLauncherDialog.
    * @param aNewFileLocation Location where the content should be saved
    */
   void saveToDisk(in nsIFile aNewFileLocation, in boolean aRememberThisPreference);
 
   /**
    * Use aApplication to launch with this content.
--- mozilla/uriloader/exthandler/nsExternalHelperAppService.cpp.orig	2007-03-07 16:23:01.000000000 -0400
+++ mozilla/uriloader/exthandler/nsExternalHelperAppService.cpp	2007-03-07 16:23:29.000000000 -0400
@@ -2187,16 +2187,22 @@ NS_IMETHODIMP nsExternalAppHandler::GetS
 }
 
 NS_IMETHODIMP nsExternalAppHandler::GetSuggestedFileName(nsAString& aSuggestedFileName)
 {
   aSuggestedFileName = mSuggestedFileName;
   return NS_OK;
 }
 
+NS_IMETHODIMP nsExternalAppHandler::GetContentLength(PRInt64 *aContentLength)
+{
+  *aContentLength = mContentLength;
+  return NS_OK;
+}
+
 nsresult nsExternalAppHandler::InitializeDownload(nsITransfer* aTransfer)
 {
   nsresult rv;
   
   nsCOMPtr<nsIURI> target;
   rv = NS_NewFileURI(getter_AddRefs(target), mFinalFileDestination);
   if (NS_FAILED(rv)) return rv;
   
